pkg/transport/transport_test.go

```diff
@@ -64,34 +86,140 @@ func TestDialNode(t *testing.T) {
 			assert.Error(t, err, tag)
 			assert.Nil(t, conn, tag)
 		}
-	}
+	})
 
-	{ // DialNode with valid target
-		timedCtx, cancel := context.WithTimeout(ctx, time.Second)
-		conn, err := client.DialNode(timedCtx, &pb.Node{
+	t.Run("DialNode with valid target", func(t *testing.T) {
+		target := &pb.Node{
 			Id: planet.StorageNodes[1].ID(),
 			Address: &pb.NodeAddress{
 				Transport: pb.NodeTransport_TCP_TLS_GRPC,
 				Address:   planet.StorageNodes[1].Addr(),
 			},
 			Type: pb.NodeType_STORAGE,
-		})
+		}
+
+		timedCtx, cancel := context.WithTimeout(ctx, time.Second)
+		dialOption := opts.DialOption(storj.NodeID{})
+		conn, err := client.DialNode(timedCtx, target, dialOption)
 		cancel()
 
 		assert.NoError(t, err)
-		assert.NotNil(t, conn)
+		require.NotNil(t, conn)
 
 		assert.NoError(t, conn.Close())
-	}
+	})
+
+	t.Run("DialNode with bad client certificate", func(t *testing.T) {
+		target := &pb.Node{
+			Id: planet.StorageNodes[1].ID(),
+			Address: &pb.NodeAddress{
+				Transport: pb.NodeTransport_TCP_TLS_GRPC,
+				Address:   planet.StorageNodes[1].Addr(),
+			},
+			Type: pb.NodeType_STORAGE,
+		}
+
+		timedCtx, cancel := context.WithTimeout(ctx, time.Second)
+		dialOption := unsignedClientOpts.DialOption(storj.NodeID{})
+		conn, err := client.DialNode(
+			timedCtx, target, dialOption,
+		)
+		cancel()
+
+		tag := fmt.Sprintf("%+v", target)
+		assert.Nil(t, conn, tag)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "bad certificate")
+	})
 
-	{ // DialAddress with valid address
+	t.Run("DialAddress with bad client certificate", func(t *testing.T) {
+		timedCtx, cancel := context.WithTimeout(ctx, time.Second)
+		dialOption := unsignedClientOpts.DialOption(storj.NodeID{})
+		conn, err := client.DialAddress(
+			timedCtx, planet.StorageNodes[1].Addr(), dialOption,
+		)
+		cancel()
+
+		assert.Nil(t, conn)
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "bad certificate")
+	})
+
+	t.Run("DialAddress with valid address", func(t *testing.T) {
 		timedCtx, cancel := context.WithTimeout(ctx, time.Second)
 		conn, err := client.DialAddress(timedCtx, planet.StorageNodes[1].Addr())
 		cancel()
 
 		assert.NoError(t, err)
-		assert.NotNil(t, conn)
-
+		require.NotNil(t, conn)
 		assert.NoError(t, conn.Close())
+	})
+}
+
+func TestDialNode_BadServerCertificate(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.NewCustom(
+		zap.L(),
+		testplanet.Config{
+			SatelliteCount:   0,
+			StorageNodeCount: 2,
+			UplinkCount:      0,
+			Reconfigure:      testplanet.DisablePeerCAWhitelist,
+			Identities:       testplanet.NewPregeneratedIdentities(),
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
 	}
+	defer ctx.Check(planet.Shutdown)
+
+	whitelistPath, err := planet.WriteWhitelist()
+	require.NoError(t, err)
+
+	planet.Start(ctx)
+
+	client := planet.StorageNodes[0].Transport
+	signedIdentities := testplanet.NewPregeneratedSignedIdentities()
+	ident, err := signedIdentities.NewIdentity()
```

Comment:
`testplanet.PregeneratedSignedIdentity(0)`

Reply:
:+1:

Reply:
I will eventually stop doing this :p